relay-server/src/envelope.rs

```diff
@@ -136,32 +139,88 @@ impl ContentType {
             Self::OctetStream => "application/octet-stream",
             Self::Minidump => "application/x-dmp",
             Self::Xml => "text/xml",
+            Self::Envelope => self::CONTENT_TYPE,
             Self::Other(ref other) => &other,
         }
     }
 
-    fn from_str(content_type: &str) -> Option<Self> {
-        match content_type {
-            "text/plain" => Some(Self::Text),
-            "application/json" => Some(Self::Json),
-            "application/x-msgpack" => Some(Self::MsgPack),
-            "application/x-dmp" => Some(Self::Minidump),
-            "text/xml" | "application/xml" => Some(Self::Xml),
-            "application/octet-stream" => Some(Self::OctetStream),
-            _ => None,
+    fn from_str(ct: &str) -> Option<Self> {
+        if ct.eq_ignore_ascii_case(Self::Text.as_str()) {
+            Some(Self::Text)
+        } else if ct.eq_ignore_ascii_case(Self::Json.as_str()) {
+            Some(Self::Json)
+        } else if ct.eq_ignore_ascii_case(Self::MsgPack.as_str()) {
+            Some(Self::MsgPack)
+        } else if ct.eq_ignore_ascii_case(Self::OctetStream.as_str()) {
+            Some(Self::OctetStream)
+        } else if ct.eq_ignore_ascii_case(Self::Minidump.as_str()) {
+            Some(Self::Minidump)
+        } else if ct.eq_ignore_ascii_case(Self::Xml.as_str())
+            || ct.eq_ignore_ascii_case("application/xml")
+        {
+            Some(Self::Xml)
+        } else if ct.eq_ignore_ascii_case(Self::Envelope.as_str()) {
+            Some(Self::Envelope)
+        } else {
+            None
         }
```

Comment:
Why do we `return None` ( and in the function `Option<ContentType>` ) when we could `return Self::Other(ct.make_ascii_lowercase())` and have the function always succeed with some `ContentType`?

Also this would simplify `impl From<&'_ str> for ContentType` which does that anyway (i.e. create an `Other(String)` for unknown content type).  and the `PartialEq<str>` implementation

Reply:
This is to optimize the implementations of `From<&str>` and `From<String>` so that we avoid allocations where possible.